### ISSUE TITLE
Unneeded comparison of upper bound of array when processing s2 more than twice as long as s1.

### ISSUE THREAD

**Issue by stuartbale** [jordanthomas/jaro-winkler#3]
I wonder if this:
`var high = (i + range <= s2.length) ? (i + range) : (s2.length - 1);`
should be:
`var high = (i + range < s2.length) ? (i + range) : (s2.length - 1);`

**Comment by jordanthomas** [comment]
Thanks for the report - I'll take a closer look this evening.

**Comment by jordanthomas** [comment]
@stuartbale Can you make a test case to recreate this? I tried this (below) and I'm still passing.

```
it('should work no matter which string is longer', function() {
  var dist = distance('short', 'somethingmuchlonger');
  var reverseDist = distance('somethingmuchlonger', 'short');
  assert.equal(dist, reverseDist);
});
```

**Comment by stuartbale** [comment]
OK - so the function still works, as on this line:
`if (s1Matches[i] !== true && s2Matches[j] !== true && s1[i] === s2[j]) {`

s2[j] returns undefined, as it is outside the upper bound on the array.
Basically, the code loops one more time than it needs to, when s2 is more than twice the length of s1.

I will change the heading of this issue to more accurately describe the scenario.

**Comment by jordanthomas** [comment]
@stuartbale Sorry for the delay, I was on vacation last week! Confirmed what you're seeing and fixed it in 0.2.7. Thanks for helping out!
